keras/optimizers/optimizer_experimental/adamw.py

```diff
@@ -173,7 +181,7 @@ def update_step(self, gradient, variable):
     alpha = (lr * tf.sqrt(1 - beta_2_power) / (1 - beta_1_power))
 
     # Apply step weight decay
-    if self.weight_decay != 0:
+    if self.weight_decay and self._do_use_weight_decay(variable):
```

Comment:
The better than used `if self.weight_decay != 0:` is `if self.weight_decay:` becouse user can set weight_decay=None and it's not a number (float) like 0, but it's empty object.